src/Orckestra.Composer.Website/App_Insights/TelemetryInitializer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Configuration;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+namespace Orckestra.Composer.Website.App_Insights
+{
+    public class TelemetryInitializer : ITelemetryInitializer
+    {
+        public const string C1Function = nameof(C1Function);
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            TelemetryConfiguration.Active.InstrumentationKey = ConfigurationManager.AppSettings["InstrumentationKey"];
```

Comment:
Having an AppSetting value for instrumentation key, and setting it with code may not be needed at all.
Most likely, if you create an environment variable  "APPINSIGHTS_INSTRUMENTATIONKEY", it will be picked up by the App Insights code.

If you want to have the setting for local deployments, you can add a check for an empty value, so it doesn't override the default behavior:

```
string instrKeyFromAppSetting = ConfigurationManager.AppSettings["InstrumentationKey"];
if (!string.IsNullOrWhitespace(instrKeyFromAppSetting)
{
    TelemetryConfiguration.Active.InstrumentationKey = instrKeyFromAppSetting;
}
```

Also - this code should be called only once, so it is better to move it either to a static constructor or a "startup handler"